docs/madr/decisions/003-logging.md

```diff
@@ -0,0 +1,245 @@
+# Logging standard
+
+* Status: accepted
+
+## Context and Problem Statement
+
+Seamless troubleshooting is a feature. We should keep more attention to the logging in Kuma.
+
+## Considered Options
+
+* Adopting consistent logging standard.
+
+## Decision Outcome
+
+Chosen option: "Adopting consistent logging standard", because it's the only option.
+
+### Positive Consequences
+
+* Better UX for users.
+* Better troubleshooting means both users and maintainers spent less time debugging an issue.
+
+### Negative Consequences
+
+* More effort to try to follow the guidelines.
+
+## Pros and Cons of the Options
+
+### Adopting consistent logging standard
+
+#### Convention
+
+Ideally, all those points are linted.
+
+1) Logs are lowercase. Can be uppercase when you start with proper noun.
+_GOOD_
+```go
+log.Info("something has happened")
+log.Info("Envoy is starting")
+```
+_BAD_
+```go
+log.Info("Something has happened")
+```
+
+2) Do not include `.` at the end of the sentence if this is the only sentence in the log
+_GOOD_
+```go
+log.Info("something has happened")
+log.Info("something has happened. Restarting the process")
+```
+_BAD_
+```go
+log.Info("something has happened.")
+```
+
+3) Either log or return an error. Don't do both, unless you want to log more details on DEBUG level.
+_GOOD_
+```go
+if err != nil {
+	return errors.Wrap(err, "more info")
+}
+...
+if err != nil {
+	log.Error(err, "more info")
+	return nil
+}
+...
+if err != nil {
+	log.V(1).Info("much more details about the error")
+    return errors.Wrap(err, "more info")
+}
+```
+_BAD_
+```go
+if err != nil {
+    log.Error(err, "more info")
+	return errors.Wrap(err, "more info")
+}
+```
+
+4) Use structured logging instead of `fmt.Sprintf`
```

Comment:
Maybe this should be even saying: "the first arg should always be a constant string". String concat, function calls etc should be used